File: sj/46week/Remove All Adjacent Duplicates In String.js
/**
 * @param {string} S
 * @return {string}
 */
var removeDuplicates = function(S) {
    
    const res = [];
    
    for (const c of S)
        {
            res[res.length-1]=== c ? res.pop() : res.push(c);
        }
    
    return res.join('');
    
};

File: jw/week46/RemoveAllAdjacentDuplicatesInString.java
class Solution {
    public String removeDuplicates(String S) {
        Stack<Character> st = new Stack<>();
        for(char c : S.toCharArray()) {
            if(st.empty()) st.push(c);
            else {
                if(st.peek() == c) st.pop();
                else st.push(c);
            }
        }
        StringBuilder sb = new StringBuilder();
        while(!st.isEmpty()) {
            sb.append(st.pop());
        }
        return sb.reverse().toString();
    }
}
